lang/types/value.go

```diff
@@ -65,17 +70,23 @@ func ValueOfGolang(i interface{}) (Value, error) {
 // a resource field. This is done with our "elvis" operator. When using this
 // function, if you pass in something with a nil value, then expect a panic or
 // an error if you're lucky.
-// FIXME: we should make sure we error instead of ever panic-ing on a nil value
-func ValueOf(v reflect.Value) (Value, error) {
-	value := v
+func ValueOf(value reflect.Value) (Value, error) {
+	// Gracefully handle invalid values instead of panic().
+	if !value.IsValid() {
+		return nil, &reflect.ValueError{Method: "types.ValueOf", Kind: reflect.Invalid}
+	}
+
 	typ := value.Type()
 	kind := typ.Kind()
+
 	for kind == reflect.Ptr {
-		typ = typ.Elem() // un-nest one pointer
-		kind = typ.Kind()
+		// Prevent panic() if value is a nil pointer and return an error.
+		if value.IsNil() {
+			return nil, ErrNilValue
+		}
 
 		// un-nest value from pointer
-		value = value.Elem() // XXX: is this correct?
```

Comment:
Why is this changing please?

Reply:
I'll add a note in the commit message. The functionality is ~ the same but the recursive call allows making the IsValid() check above on every pointer dereference. I'm not 100% if we need to do it (if the stdlib can ever return invalid reflect.Value values, that is) but I figured better safe than sorry. It's a pretty cheap check

Reply:
Reverted this back as it's not required